Add strict param to OsvSources

tmihordea opened the pull request:
This is useful for schema evolution. For example if we add a new field in a ScaldingType model we will be able to use the new code to read older files. In this case the value of the new field will be the default set in the ScaldingType or null in case there is no default.

jqcoffey commented:
While you're at it do you want to do the same thing for TsvSources?  It's effectively the same thing with a different separator.  There's probably a reasonable opportunity to refactor a bit as well, if you're feeling ambitious.

Commit:
Add strict param to OsvSources

This is useful for schema evolution. For example if we add a new field in a ScaldingType model we will be able to use the new code to read older files. In this case the value of the new field will be the default set in the ScaldingType or null in case there is no default.
